Documentation/networking/dns.md

```diff
@@ -9,25 +21,91 @@ $ sudo rkt run --dns=8.8.8.8 pod.aci
 Other parameters can be given:
 
 ```bash
-$ sudo rkt run --dns=8.8.8.8 --dns=8.8.4.4 --dns-search=foo.com --dns-opt=debug pod.aci
+$ sudo rkt run \
+	--dns=8.8.8.8 --dns=4.2.2.2 \
+	--dns-domain=example.org \
+	--dns-opt=debug --dns-opt=rotate \
+	--dns-search=example.com --dns-search=example.gov \
+	pod.aci
 ```
 
 This will generate the following `/etc/resolv.conf` for the applications:
 
 ```
-# Generated by rkt
+# Generated by rkt run
 
-search foo.com
+search example.com example.gov
 nameserver 8.8.8.8
-nameserver 8.8.4.4
-options debug
+nameserver 4.2.2.2
+options debug rotate
+domain example.org
 ```
 
+### "Magic" parameters
+
+#### `host`
+The magic parameter `host` will bind-mount the host's `/etc/resolv.conf` in to the applications.
+This will be a read-only mount.
+
+#### `none`
+The magic parameter `none` will ignore any DNS configuration from CNI. This will ensure that
+the image's `/etc/resolv.conf` has precedence.
+
+### Precedence
+`resolv.conf` can be generated by multiple components. The order of precedence is:
+
+1. If `--dns`, et al. are passed to `rkt run`
+2. If a CNI plugin returns DNS information, unless `--dns=none` is passed
+3. If a volume is mounted on `/etc/resolv.conf`
+4. If the application container includes `/etc/resolv.conf`
+
+
+
+## `/etc/hosts`
+`rkt run` provides one option with two modes:
+
+* `--hosts-entry <IP>=<HOST>`  
+* `--hosts-entry host`
+
+Passing `--hosts-entry=host` will bind-mount (read-only) the hosts's `/etc/hosts`
+in to every application.
+
+When passing IP=HOST pairs:
+
+```bash
+$ rkt run ... --hosts-entry 198.51.100.0=host1,198.51.100.1=host2 --hosts-entry 198.51.100.0=host3
+```
+
+creates
+
+```
+< some default localhost entries...>
+
+198.51.100.0 host1 host3
+198.51.100.1 host2
+```
+
+
+
+### Precedence
+1. If `--hosts-entry` is passed to `rkt run`
+2. If a volume is mounted on `/etc/hosts`
+3. If the application container includes `/etc/hosts`
+4. Otherwise, a fallback stub `/etc/hosts`  is created
+
+
+
+## Example
 The following example shows that the DNS options allow the pod to resolve names successfully:
 
-```sh
-$ sudo rkt run --dns 8.8.8.8  --net=host kinvolk.io/aci/busybox:1.24 --exec /bin/busybox -- ping -c1 coreos.com
+```
```

Comment:
`bash` was an error (should have been `sh`). `sh` was removed here because the output is not actually a shell script.